### PR TITLE
[ci] Refactor ci

### PR DESCRIPTION
**Pull request by alvicsam**
In the PR:
- Removed `generateDockerfile` script in favor of multistage Dockerfile.
- Changed `.gtilab-ci.yml` so it looks like in other projects
- Added `helm-stg` to deploy the bot to staging (if tests on staging succeed we'll move it to `parity-prod`)
- Added logic to deploy on stage: on master, on tags `stg-v*` and `v*`

cc https://github.com/paritytech/ci_cd/issues/564

### CODE REVIEW

**Review by mordamax — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,85 +1,109 @@
...
-  IMAGE_REPOSITORY: docker.io/paritytech/command-bot
-  CI_IMAGE: ${IMAGE_REPOSITORY}:ci
-  KUBE_NAMESPACE: try-runtime
-  DEPLOYMENT_TAG: ${CI_ENVIRONMENT_NAME}_${CI_COMMIT_TAG}
-  DATA_PATH: /data
-  PING_PORT: 3001
-  TASK_DB_VERSION: v3
+  IMAGE_NAME:                      docker.io/paritytech/command-bot
+  CI_IMAGE:                        ${IMAGE_NAME}:ci
+  KUBE_NAMESPACE:                  try-runtime
```
[the remark this answers is not in the document]
Comment (alvicsam) [reply .gitlab-ci.yml:10]: Absolutely, but in this case it should be renamed in the cluster `parity-chains` and I don't have access there. If we can prod move to `parity-prod` in the future then I'll rename the namespace

[the remark this answers is not in the document]
Comment (mordamax) [reply .gitlab-ci.yml:10]: if we can test try-runtime on staging on parity-stg then potentially it means that we can move and rename, right?

[the remark this answers is not in the document]
Comment (alvicsam) [reply .gitlab-ci.yml:10]: Yes, that's correct. I will try deploying to `command-bot` namespace during refactor

**Review by mutantcornholio — APPROVED** [review]
♥️

**Comment by mordamax** [comment]
can we delete scripts/generateDockerfile and move scripts/Dockerfile to root?

**Review by sergejparity — APPROVED** [review]

**Pull request merged**